frontend/dockerfile/dockerfile2llb/directives.go

```diff
@@ -20,10 +34,22 @@ type Directive struct {
 }
 
 func DetectSyntax(r io.Reader) (string, string, []parser.Range, bool) {
-	directives := ParseDirectives(r)
-	if len(directives) == 0 {
+	bs, err := io.ReadAll(r)
```

Comment:
If this needs to do full memory allocation again then change the input to `[]byte`